lib/pure/collections/sequtils.nim

```diff
@@ -1106,16 +1122,12 @@ when isMainModule:
     doAssert newSeq[int](0).mapIt(it) == @[]
 
   block: # mapIt redifinition check, see https://github.com/nim-lang/Nim/issues/8580
-    let t = [1,2].mapIt(it)
-    doAssert t == @[1,2]
+    let s2 = [1,2].mapIt(it)
+    doAssert s2 == @[1,2]
 
   block:
-    var counter = 0
-    proc getInput():auto =
-      counter.inc
-      [1, 2]
-    doAssert getInput().mapIt(it*2).mapIt(it*10) == @[20, 40]
-    # make sure argument evaluated only once, analog to
+    counter = 0
+    doAssert [1,2].identity().mapIt(it*2).mapIt(it*10) == @[20, 40]
```

Comment:
simplifies test case, reusing `identity` and `counter`, as done in other test cases in this file